Framework/API/src/FileFinder.cpp

```diff
@@ -545,16 +545,18 @@ void FileFinderImpl::getUniqueExtensions(
 
 /**
  * Performs validation on the search text entered into the File Finder. It will
- * throw an exception if a problem is found. This error would then be caught and
- * displayed to a user.
+ * return a vector of error messages if problems are found.
  * @param searchText :: The text to validate.
- * @throw std::invalid_argument if the searchText is invalid
+ * @return A vector of error messages.
  */
-void FileFinderImpl::validateRuns(const std::string &searchText) const {
+std::vector<std::string>
+FileFinderImpl::validateRuns(const std::string &searchText) const {
+  std::vector<std::string> errors;
```

Comment:
Quick Q. Any reason we can't just return a string here? It also has an `.empty` method..